cumulus/xcm/xcm-emulator/src/lib.rs

```diff
@@ -576,6 +589,8 @@ macro_rules! decl_test_parachains {
 				pallets = {
 					$($pallet_name:ident: $pallet_path:path,)*
 				}
+				$(on_initialize = $on_initialize_fn:expr,)?
+				$(on_finalize = $on_finalize_fn:expr,)?
```

Comment:
These expresions are wrong. It would be in any case something like:
```rust
on_initialize = $( $on_initialize_fn:expr,)?
$(on_finalize = $on_finalize_fn:expr,)?
```

But we do not really want to pass many `on_initialize_fn`, `on_finalize_fn`. Just a generic single function that is expecting a `BlockNumber` as argument. When you define you function you can call as many `palle_whatever::on_initialize(blockl_number)` you want inside it.